src/API/Site/Controllers/TourController.php

```diff
@@ -0,0 +1,171 @@
+			'/tours/(?P<id>[a-z]+)',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_tours_read_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			],
+		);
+
+		/**
+		 * POST Update the tour visualizations
+		 */
+		$this->register_route(
+			'/tours',
+			[
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->get_tours_create_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_schema_properties(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			],
+		);
+	}
+
+	/**
+	 * Callback function for returning the tours
+	 *
+	 * @return callable
+	 */
+	protected function get_tours_read_callback(): callable {
+		return function ( Request $request ) {
+			try {
+				$tour_id = $request->get_url_params()['id'];
+				return $this->prepare_item_for_response( $this->get_tour( $tour_id ), $request );
+			} catch ( Exception $e ) {
+				return $this->response_from_exception( $e );
+			}
+		};
+	}
+
+	/**
+	 * Callback function for saving the Tours
+	 *
+	 * @return callable
+	 */
+	protected function get_tours_create_callback(): callable {
+		return function( Request $request ) {
+			try {
+				$tour_id           = $request->get_param( 'id' );
+				$tours             = $this->get_tours();
+				$tours[ $tour_id ] = $this->prepare_item_for_database( $request );
+
+				if ( $this->options->update( OptionsInterface::TOURS, $tours ) ) {
+					return new Response(
+						[
+							'status'  => 'success',
+							'message' => __( 'Successfully updated the tour.', 'google-listings-and-ads' ),
+						],
+						200
+					);
+				} else {
+					throw new Exception( __( 'Unable to updated the tour.', 'google-listings-and-ads' ), 400 );
+				}
+			} catch ( Exception $e ) {
+				return $this->response_from_exception( $e );
+			}
+		};
+	}
+
+	/**
+	 * Get the tours
+	 *
+	 * @return array|null The tours saved in databse
+	 */
+	private function get_tours(): ?array {
+		return $this->options->get( OptionsInterface::TOURS );
+	}
+
+	/**
+	 * Get the tour by Id
+	 *
+	 * @param string $tour_id The tour ID
+	 * @return array The tour
+	 * @throws Exception In case the tour is not found.
+	 */
+	private function get_tour( string $tour_id ): array {
+		$tours = $this->get_tours();
+		if ( ! isset( $tours[ $tour_id ] ) ) {
+			throw new Exception( __( 'Tour not found', 'google-listings-and-ads' ), 404 );
+		}
+
+		return $tours[ $tour_id ];
+	}
+
+	/**
+	 * Get the item schema properties for the controller.
+	 *
+	 * @return array The Schema properties
+	 */
+	protected function get_schema_properties(): array {
+		return [
+			'id'      => [
+				'description'       => __( 'The Id for the tour.', 'google-listings-and-ads' ),
+				'type'              => 'string',
```

Comment:
💅  We should add a pattern validation for `POST /tours` to align with the API `GET /tours/(?P<id>[a-z]+)`. If we want to restrict the ID to be only letters it would be great adding a validation here so the ID can be verified when creating a tour.

```suggestion
				'type'              => 'string',
				'pattern'           => '^[a-z]+$',
```

Reply:
Thanks. I added [a-zA-Z0-9-_]